libs/go/libp2p_node/utils/utils_test.go

```diff
@@ -0,0 +1,160 @@
+/* -*- coding: utf-8 -*-
+* ------------------------------------------------------------------------------
+*
+*   Copyright 2018-2021 Fetch.AI Limited
+*
+*   Licensed under the Apache License, Version 2.0 (the "License");
+*   you may not use this file except in compliance with the License.
+*   You may obtain a copy of the License at
+*
+*       http://www.apache.org/licenses/LICENSE-2.0
+*
+*   Unless required by applicable law or agreed to in writing, software
+*   distributed under the License is distributed on an "AS IS" BASIS,
+*   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*   See the License for the specific language governing permissions and
+*   limitations under the License.
+*
+* ------------------------------------------------------------------------------
+ */
+
+package utils
+
+import (
+	"encoding/json"
+	"bytes"
+	"testing"
+	"errors"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/assert"
+	mocks "libp2p_node/mocks"
+	gomock "github.com/golang/mock/gomock"
+
+)
+
+// Crypto operations
+
+
+func TestEthereumCrypto(t *testing.T) {
+	//privateKey := "0xb60fe8027fb82f1a1bd6b8e66d4400f858989a2c67428a4e7f589441700339b0"
+	publicKey := "0xf753e5a9e2368e97f4db869a0d956d3ffb64672d6392670572906c786b5712ada13b6bff882951b3ba3dd65bdacc915c2b532efc3f183aa44657205c6c337225"
+	address := "0xb8d8c62d4a1999b7aea0aebBD5020244a4a9bAD8"
+	publicKeySignature := "0x304c2ba4ae7fa71295bfc2920b9c1268d574d65531f1f4d2117fc1439a45310c37ab75085a9df2a4169a4d47982b330a4387b1ded0c8881b030629db30bbaf3a1c"
+
+	addFromPublicKey, err := EthereumAddressFromPublicKey(publicKey)
+	if err != nil || addFromPublicKey != address {
+		t.Error(
```

Comment:
Shouldn't this be `Fatal`? You both want the test to fail, but also doesn't make sense to continue, as `Error` allows?

Reply:
also below

Reply:
it will fail anyway, but can be changed to Fatal as some asserts can be changed to required